src/components/data-management/ProjectDetails.jsx

```diff
@@ -551,15 +534,14 @@ const ProjectDetails = ({ width, height }) => {
     if (!uploadDetailsModalDataRef.current) {
       return;
     }
-
     const { sampleUuid, file } = uploadDetailsModalDataRef.current;
-
-    const bucketKey = `${activeProjectUuid}/${sampleUuid}/${file.name}`;
+    const name = file.name || bundleToUpload.name;
```

Comment:
Similar to below review comment, a comment here would be good to explain when each path is taken.